calculate_average_yourwass.sh

```diff
@@ -19,5 +19,8 @@
 # source "$HOME/.sdkman/bin/sdkman-init.sh"
 # sdk use java 21.0.1-graal 1>&2
 
-JAVA_OPTS="--enable-preview --enable-native-access=ALL-UNNAMED --add-modules jdk.incubator.vector"
-java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_yourwass
+JAVA_OPTS="-Xlog:all=off -Djdk.incubator.vector.VECTOR_ACCESS_OOB_CHECK=0 --enable-preview --enable-native-access=ALL-UNNAMED --add-modules jdk.incubator.vector"
+
+eval "exec 3< <({ java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_yourwass; })"
+read <&3 result
```

Comment:
What's the purpose of this eval and read construct here?

Reply:
Well copy pasting in a hurry, copies bad code. Although the idea is good, the implementation by serkan-ozal that I copied was sloppy.                           
All the eval things and opening a 3rd file descriptor are not really needed. The code is equivalent to and has the same timings with:                        
                                                                                
```
read result < <( java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_yourwass )
echo -e "$result"  
```                                                                                                                                             
Now, to explain this: **read** reads **one** line from the output of the java program that is redirected into the variable **result**, that's it. It just doesn't wait for a second line, or whatever the java program does afterwards. It immediately prints the result. Thus, also the `System.out.close()` in the java code, is not really needed.

The unneeded eval evaluates the string. When it does, it opens a 3rd file descriptor and redirects the output of the java command onto it. Then the read reads from that file descriptor one line into the variable result.

Reply:
Ah, got it. Neat trick :)